### PR TITLE
Cr 1763 i pv6

### PR DESCRIPTION
**Pull request by Paul-Joel**
Tweaks to IP extraction to cope with IPv6 values, and update/split of unit tests

Signed-off-by: Paul-Joel <paul.joel@ons.gov.uk>

# Motivation and Context
Tweaks to IP extraction to cope with IPv6 values, and update/split of unit tests

# What has changed
Changes to ip split and associated unit tests

No Cucumber updates required

# How to test?

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1763

### CODE REVIEW

**Review by rpcatling — APPROVED** [review]
LGTM

File: app/utils.py [hunk 1]
```diff
@@ -49,9 +49,17 @@ def get_now_utc():
     def single_client_ip(request):
         if request['client_ip']:
             client_ip = request['client_ip']
-            ip_validation_pattern = re.compile(r'^[0-9.,\s]*$')
+            ip_validation_pattern = re.compile(r'^[0-9a-zA-Z.:,\s]*$')
+            single_ip_validation_pattern = re.compile(r'^[0-9.]*$')
```
Comment (MartinHumphrey) [inline app/utils.py:53]: The '*' matches 0 or more repetitions of the character set, would "+" matching 1 or more repetitions be better?

File: app/utils.py [hunk 2]
```diff
@@ -49,9 +49,17 @@ def get_now_utc():
     def single_client_ip(request):
         if request['client_ip']:
             client_ip = request['client_ip']
-            ip_validation_pattern = re.compile(r'^[0-9.,\s]*$')
+            ip_validation_pattern = re.compile(r'^[0-9a-zA-Z.:,\s]*$')
```
Comment (MartinHumphrey) [inline app/utils.py:52]: The '*' matches 0 or more repetitions of the character set, would "+" matching 1 or more repetitions be better?

File: app/utils.py (hunk 1 above)
Comment (MartinHumphrey) [inline app/utils.py:53]: The '*' matches 0 or more repetitions of the character set, would "+" matching 1 or more repetitions be better?

**Review by MartinHumphrey — APPROVED** [review]
Two comments about regular expression but don't make a significant difference.

**Review by rpcatling — CHANGES_REQUESTED** [review]
Initial regex needs to let through junk at the beginning.

**Review by PeterBochel — APPROVED** [review]

**Review by rpcatling — APPROVED** [review]

**Pull request merged**